src/core/ensemble_scheduler.cc

```diff
@@ -658,33 +660,24 @@ EnsembleContext::CheckAndSetEnsembleOutput()
     }
 
     // Use the memory type of the memory block as preferred memory type
-    TRTSERVER_Memory_Type dst_memory_type;
+    TRTSERVER_Memory_Type dst_memory_type, allocated_memory_type;
     int64_t memory_type_id;
     size_t content_size;
     memory_block->BufferAt(0, &content_size, &dst_memory_type, &memory_type_id);
 
     void* buffer;
+    int64_t allocated_memory_type_id;
     RETURN_IF_ERROR(response_provider_->AllocateOutputBuffer(
         output_pair.first, &buffer, expected_byte_size, shape, dst_memory_type,
-        memory_type_id));
+        memory_type_id, &allocated_memory_type, &allocated_memory_type_id));
```

Comment:
This is not right, either let `AllocateOutputBuffer()` write actual type into `dst_memory_type` or change the code below to condition on `allocated_memory_type` instead of `dst_memory_type`.

Reply:
Fixed.